### DISCUSSION TITLE
Topic Operator - Using MSK with AWS IAM Auth

### DISCUSSION THREAD

**Discussion by sherifkayad** [strimzi/strimzi-kafka-operator#7222] [Q&A]
Hello,

Can someone please tell me if the Topic Operator can support using an MSK cluster with AWS IAM Auth? .. I didn't manage to find anything in the documentation that suggests so, however wanted to be sure.

If that's currently not supported, then would it be on the roadmap? .. I saw some open Github issues mentioning the AWS IAM Auth for some different components of the Operator, however, again I am not sure what this means for the Topic Operator nor do I understand if that's something that the Strimzi project would prioritize / has on the Roadmap.

Thanks.

**Comment by scholzj** [comment]
Our priority is running the Kafka server components. Not providing tooling for Amazon MSK or other services. So we would not prfioritize this as part of our own work. But contributors are always welcomed, most of the efforts to make the standalone Topic or User operators run with managed service X or Y are usually user contributed. 

The way it normally works, is that:
1) You build your own container image which adds the libraries for your cloud provider's managed service
2) Contribute a PR which makes the Topic Operator configurable enough to let you configure the provider authentication (in this case, I think this would be the JAAS options for configuring SASL authentication for example)

The separation into these two steps is important because the second part is something that can be tested on its own. The integration with some particular cloud provider cannot be easily tested (without cloud accounts, budgets, credit cards etc.).
